File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        ans=0
        def deepness(node):
            nonlocal ans
            l=0
            r=0
            if(node.left):
                l=deepness(node.left)
            if(node.right):
                r=deepness(node.right)
            ans=max(l+r+1,ans)
            return max(l,r)+1
        deepness(root)
        return ans-1
            
            

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
private:
    int ans(TreeNode* node, int maxvalue){
        if(node == nullptr){
            return 0;
        }
        int add=0;
        if(node->val>= maxvalue){
            add=1;
        }
     
        return add+ans(node->left,max(node->val,maxvalue)) + ans (node->right,max(node->val,maxvalue));
    }
public:
    int goodNodes(TreeNode* root) {
        return ans(root,root->val);
    }
};
